go/interpreter/lfvm/interpreter_intergration_test.go

```diff
@@ -0,0 +1,689 @@
+	tests = addTrivialTest(tests, gas(3), stackSize(17, 17), getOpCodesInRange(SWAP1, SWAP16)...)
+
+	///////////////////////////////////
+	// Stack manipulation
+
+	// PUSH1 - PUSH32
+	tests = append(tests, generateTestCaseFor(
+		func(op OpCode) testDeclaration {
+			dataNum := int((op - PUSH1) / 2)
+			code := attachDataToOp(op, dataNum)
+			return testDeclaration{
+				code:  code,
+				stack: stackSize(0, 1),
+				gas:   cost(3),
+			}
+		},
+		getOpCodesInRange(PUSH1, PUSH32))...)
+
+	// PUSH0
+	tests = append(tests, testDeclaration{
+		code:               []Instruction{{PUSH0, 0}},
+		stack:              stackSize(0, 1),
+		revisionConstraint: validFrom(tosca.R12_Shanghai),
+		gas:                cost(2),
+	})
+	// POP
+	tests = addTrivialTest(tests, gas(2), stackSize(1, 0), POP)
+
+	///////////////////////////////////
+	// SLOAD - STORE
+
+	tests = append(tests, []testDeclaration{
+		{
+			nameOverride:       "SLOAD",
+			code:               []Instruction{{SLOAD, 0}},
+			stack:              stackSize(1, 1),
+			revisionConstraint: validBefore(tosca.R09_Berlin),
+			mockCalls: func(mock *tosca.MockRunContext) {
+				mock.EXPECT().GetStorage(tosca.Address{}, toKey(1)).Return(toWord(0))
+			},
+			gas: cost(800),
+		},
+	}...)
+
+	///////////////////////////////////
+	// Super instructions (without jump)
+
+	tests = addTrivialTest(tests, gas(4), stackSize(2, 0), POP_POP)
+	tests = addTrivialTest(tests, gas(5), stackSize(2, 1), SWAP1_POP)
+	tests = addTrivialTest(tests, gas(5), stackSize(3, 2), SWAP2_POP)
+	tests = addTrivialTest(tests, gas(6), stackSize(1, 1), PUSH1_SHL, PUSH1_ADD)
+	tests = addTrivialTest(tests, gas(6), stackSize(1, 3), PUSH1_DUP1)
+	tests = addTrivialTest(tests, gas(6), stackSize(2, 2), DUP2_LT)
+	tests = addTrivialTest(tests, gas(6), stackSize(3, 3), SWAP2_SWAP1)
+	tests = addTrivialTest(tests, gas(6), stackSize(0, 2), PUSH1_PUSH1)
+	tests = addTrivialTest(tests, gas(10), stackSize(5, 3), POP_SWAP2_SWAP1_POP)
+	tests = addTrivialTest(tests, gas(11), stackSize(4, 3), SWAP1_POP_SWAP2_SWAP1)
+	tests = addTrivialTest(tests, gas(12), stackSize(2, 1), DUP2_MSTORE)
+	tests = addTrivialTest(tests, gas(14), stackSize(5, 3), AND_SWAP1_POP_SWAP2_SWAP1)
+
+	testWithData := func(gas gas, stack stackTest, requiredDataOps int, opCodes ...OpCode) []testDeclaration {
+		tests := make([]testDeclaration, len(opCodes))
+		for i, op := range opCodes {
+			code := attachDataToOp(op, requiredDataOps)
+			tests[i] = testDeclaration{
+				code:  code,
+				stack: stack,
+				gas:   cost(gas),
+			}
+		}
+		return tests
+	}
+
+	tests = append(tests, testWithData(15, stackSize(0, 1), 1, PUSH1_PUSH1_PUSH1_SHL_SUB)...)
+	tests = append(tests, testWithData(9, stackSize(1, 4), 2, PUSH1_PUSH4_DUP3)...)
+
+	///////////////////////////////////
+	// Jump instructions
+
+	tests = append(tests, []testDeclaration{
+
+		{
+			nameOverride: "JUMP",
+			code:         []Instruction{{PUSH1, 2 << 8}, {JUMP, 0}, {JUMPDEST, 0}},
+			gas:          cost(3 + 8 + 1),
```

Comment:
I like very much this manner of describing costs, but I only know where they come from because we discussed it offline, do you think you could add a generic comment explaining how you come up with these sums ?